jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -687,6 +690,40 @@ private void set(Object obj, XmlParser.Node node) throws Exception
                     errors.add(e);
                 }
 
+                // Try a builder
+                try
+                {
+                    Method builder = oClass.getMethod(name, vClass);
```

Comment:
> I found several bugs in XmlConfiguration that prevented it being used with a Builder pattern.

what were the bugs specifically

Reply:
+ A `Set` could not be used with a builder style method (kind of missing feature more than a bug)
+ If a `property` element was used in a `Set` it could only be a string and the `type` element was ignored.
+ When trying to find a native match, the `vClass` local variable was overwritten with the native TYPE being tried. This affected subsequent match attempts using the wrong `vClass`

You can see the fixes above

Reply:
@lachlan-roberts I've moved the comment above to the PR description.